src/test/unit/services/util/mcmc_writer_test.cpp

```diff
@@ -11,10 +11,33 @@
 namespace test {
 // mock_throwing_model_in_write_array throws exception in the write_array()
 // method
-class throwing_model : public stan_model {
+class throwing_model : public stan::model::model_base_crtp<throwing_model> {
+ private:
+  int pos__;
+
  public:
-  throwing_model(stan::io::var_context& context, std::ostream* pstream)
-      : stan_model(context, pstream) {}
+  ~throwing_model() {}
+
+  std::string model_name() const final { return "throwing_model"; }
```

Comment:
Most of this new code isn't used in the tests here. Can it just get deleted? Otherwise in the future someone might think they need it all.

Reply:
Will try to trim it out yeah. Was mostly focusing on making the test run.

Reply:
I trimmed what I could. I couldn't remove a lot since most of the functions are required in the model, otherwise, we get a compile error for virtual functions from model_base that are not defined.